examples/medplum-patient-intake-demo/src/bots/core/intake-utils.ts

```diff
@@ -392,52 +466,38 @@ export function getGroupRepeatedAnswers(
   response: QuestionnaireResponse,
   groupLinkId: string
 ): Record<string, QuestionnaireResponseItemAnswer>[] {
+  // Find the questionnaire item based on groupLinkId
   const questionnaireItem = findQuestionnaireItem(questionnaire.item, groupLinkId) as QuestionnaireItem;
-  const responseItem = findQuestionnaireItem(response.item, groupLinkId) as QuestionnaireResponseItem;
 
   if (questionnaireItem.type !== 'group' || !questionnaireItem?.item) {
     return [];
   }
 
-  const responses = responseItem.item ?? [];
-  let responseCursor = 0;
+  // Get all response items corresponding to the groupLinkId
+  const responseGroups = response.item?.filter((item) => item.linkId === groupLinkId);
 
-  const linkIds = questionnaireItem.item.map((item) => item.linkId);
-  let linkCursor = 0;
-
-  const groupAnswers = [];
-  let answerGroup = {};
+  if (!responseGroups || responseGroups?.length === 0) {
+    return [];
+  }
 
-  // It expects that responses will be organized in the same order the form is filled. Eg.:
+  // It expects that responses will be organized in separate groups with the same groupLinkId. Eg.:
```

Comment:
The `getGroupRepeatedAnswers` helper was considering a wrong questionnaire response structure, I fixed it and its related code in the `intake-form-test-data.ts` file.